src/dbnode/storage/series/buffer.go

```diff
@@ -752,6 +802,8 @@ func (b *dbBufferBucket) merge() (mergeResult, error) {
 		}
 	}()
 
+	// Rank bootstrapped blocks as data that has appeared before data that
```

Comment:
hm can this be violated during a topology change?

Reply:
This is mainly a best effort change to be honest, as we know without keeping a timestamp next to each value of when it was written we can't guarantee selecting the last upserted value when reading values returned from multiple replicas.

It *should* be a good approximation however, which is what we need right now.  When we are ready to begin solving it by storing metadata next to the values we can offer configuration for either strict or best effort upserts.